### PR TITLE
Fix: graphql-transformer-migrator add modelgen flag after v2 migrate

### LINKED ISSUE #9503: amplify codegaen models - after migration to V2
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Vue, Not applicable

### Amplify APIs

GraphQL API

### Amplify Categories

api

### Environment information

<details>

```
amplify --v
7.6.5

```

</details>

## After migrating GraphQL Transformer v1 to v2, amplify models codegen models gives warnings

```
amplify codegen models
GraphQL schema compiled successfully.

Edit your schema at C:\tmp\test\amplify\backend\api\modelgentest\schema.graphql or place .graphql files in a directory at C:\tmp\test\amplify\backend\api\modelgentest\schema
warning: Model Blog has field posts of type Post but its not connected. Add a @connection directive if want to connect them.
warning: Model Post has field blog of type Blog but its not connected. Add a @connection directive if want to connect them.
warning: Model Post has field comments of type Comment but its not connected. Add a @connection directive if want to connect them.
warning: Model Comment has field post of type Post but its not connected. Add a @connection directive if want to connect them.
Successfully generated models. Generated models can be found in C:\tmp\test\src

```

### Expected behavior

`amplify codegen models ` should return correct models after migration to V2 graphQL transformers

### Reproduction steps

1. `amplify init` (select default)
2. `amplify add api `(select blog example with relationships)
3. `amplify push`
4. `amplify migrate api`
5. `amplify codegen`  (select typescript and level 2 depth)
6. `amplify codegen models` <--- This returns warnings

### Code Snippet

_No response_

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by marcvberg**
Fixes #9503

#### Description of changes
Sets the modelgen plugin flag on migration to true so that erroneous warnings are not thrown about missing connections

#### Description of how you validated changes
Ran a local migration

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cjihrig — APPROVED** [review]

**Review by lazpavel — APPROVED** [review]

File: packages/amplify-graphql-transformer-migrator/src/state-migrator.ts [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
 };
 
-export const revertTransformerVersion = async (env?: string): Promise<void> => {
-  const mutation = (cliJSON: $TSAny) => {
-    _.set(cliJSON, ['features', 'graphqltransformer', 'useexperimentalpipelinedtransformer'], false);
-    _.set(cliJSON, ['features', 'graphqltransformer', 'transformerversion'], 1);
-    _.set(cliJSON, ['features', 'graphqltransformer', 'suppressschemamigrationprompt'], false);
-  };
+export const backupCliJson = async (resourceDir: string, env? :string): Promise<void> => {
+  let cliJson = getCliJsonFile(env);
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-migrator/src/state-migrator.ts:19]:
```suggestion
  const cliJson = getCliJsonFile(env);
```

File: packages/amplify-graphql-transformer-migrator/src/state-migrator.ts [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+  let backupPath = path.join(backupLocation(resourceDir), 'cli.json');
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-migrator/src/state-migrator.ts:20]:
```suggestion
  const backupPath = path.join(backupLocation(resourceDir), 'cli.json');
```

File: packages/amplify-graphql-transformer-migrator/src/state-migrator.ts [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+  JSONUtilities.writeJson(backupPath, cliJson);
+};
+
+export const revertTransformerVersion = async (resourceDir: string, env?: string): Promise<void> => {
+  let backupPath = path.join(backupLocation(resourceDir), 'cli.json');
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-migrator/src/state-migrator.ts:25]:
```suggestion
  const backupPath = path.join(backupLocation(resourceDir), 'cli.json');
```

File: packages/amplify-graphql-transformer-migrator/src/state-migrator.ts [hunk 4]
```diff
@@ -1,22 +1,34 @@
...
+  let backupJson: $TSAny = JSONUtilities.readJson(backupPath);
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-migrator/src/state-migrator.ts:26]:
```suggestion
  const backupJson: $TSAny = JSONUtilities.readJson(backupPath);
```

File: packages/amplify-graphql-transformer-migrator/src/state-migrator.ts [hunk 5]
```diff
@@ -34,3 +46,15 @@ const mutateCliJsonFile = async (mutation: (cliObj: $TSAny) => void, env?: strin
...
+const getCliJsonFile = (env?: string): Promise<$TSAny> => {
+  const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+  let cliJSON;
+  if (env) {
+    cliJSON = stateManager.getCLIJSON(projectPath, env, { throwIfNotExist: false });
+  }
+  if (!cliJSON) {
+    cliJSON = stateManager.getCLIJSON(projectPath);
+  }
+  return cliJSON;
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-migrator/src/state-migrator.ts:59]:
```suggestion
  return cliJSON ?? stateManager.getCLIJSON(projectPath);
```

**Review by cjihrig — CHANGES_REQUESTED** [review]
Since this PR has changed a good bit since it's original version, can you include some tests as well.

**Review by cjihrig — APPROVED** [review]

**Pull request merged**